src/pods/agent/TemplateAgent.ts

```diff
@@ -0,0 +1,15 @@
+import type { Agent } from './Agent';
+
+/**
+ * An {@link Agent} that contains a reference to a template.
+ */
+export interface TemplateAgent extends Agent {
+  template: string;
```

Comment:
Yea probably makes more sense. I originally had this class extend Dict, but since that moved to Agent as well there's no real reason to have this one.